crates/project_panel/src/project_panel.rs

```diff
@@ -101,6 +101,7 @@ pub struct ProjectPanel {
     // We keep track of the mouse down state on entries so we don't flash the UI
     // in case a user clicks to open a file.
     mouse_down: bool,
+    hovered_entries: HashSet<ProjectEntryId>,
```

Comment:
Do the listener callbacks run potentially on a different thread and this HashSet should be threadsafe?

Reply:
Yeah, listeners run on foreground IIRC, so we should be good here. Besides, it wouldn't compile if it wasn't thread-safe 😄